tests/test_integration.py

```diff
@@ -88,6 +104,40 @@ def setUpModule():
 
     query.commit()
 
+    # Cancelled builds are forcefully killed by Jenkins. They did not have the
+    # opportunity to clean up so we forcibly destroy everything found on any HV
+    # which would interrupt our work in the current JENKINS_EXECUTOR.
+    for hv in HYPERVISORS:
+        hv.get_storage_pool().refresh()
+        for domain in hv.conn().listAllDomains():
+            if match(('^([0-9]+_)?' + VM_HOSTNAME_PATTERN + '$').format(
```

Comment:
* You don't want to check for VM_HOSTNAME_PATTERN = 'igvm-{}-{}.test.ig.local' do you ? 
* How about parallel running builds ?